### PR TITLE
Support to get remapped service name

### PR DESCRIPTION
**Pull request by Barry-Xu-2018**
Address #954  

`srv_name` in service & client save real service name (remapped name).

### CODE REVIEW

**Comment by Barry-Xu-2018** [comment]
`srv_name` isn't set as the property of service & client.   
So maybe it is better to make new property `service_name` for service & client class to save real service name. `srv_name` only save service name input by user.

File: rclpy/rclpy/node.py [hunk 1]
```diff
@@ -1622,7 +1622,7 @@ def create_client(
 
         client = Client(
             self.context,
-            client_impl, srv_type, srv_name, qos_profile,
+            client_impl, srv_type, client_impl.service_name, qos_profile,
```
Comment (fujitatomoya) [inline rclpy/rclpy/node.py:1625]: this changes the current behavior but i think the intention is to store the actual name with implementation. so i think changing into `client_impl.service_name` is the correct to address the bug.

**Review by fujitatomoya — APPROVED** [review]
lgtm

**Comment by fujitatomoya** [comment]
@iuhilnehc-ynos can i use another review? if you are good to go, please start the CI.

**Comment by iuhilnehc-ynos** [comment]
> `srv_name` isn't set as the property of service & client. So maybe it is better to make new property `service_name` for service & client class to save real service name. `srv_name` only save service name input by user.

Yeah. This style is also what I expected. Please see
https://github.com/ros2/rclpy/blob/1843c2259260787d94975f658ed195a82f6eb90c/rclpy/rclpy/publisher.py#L81-L84

I prefer to use a new property named 'service_name' for the Service and Client.

BTW. I have no idea if users want to use the original `srv_name` passed into Service and Client even though there aren't relative names in the Publisher and Subscriber. Let's just keep it.

**Comment by Barry-Xu-2018** [comment]
@iuhilnehc-ynos Thanks for your comments.   
I will update codes to use new property `service_name` for service & client class.

**Review by iuhilnehc-ynos — APPROVED** [review]

**Comment by iuhilnehc-ynos** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19419/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13947/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20140/)

**Pull request merged**